internal/querycoordv2/observers/resource_observer.go

```diff
@@ -105,9 +105,10 @@ func (ob *ResourceObserver) checkResourceGroup() {
 						zap.Error(err),
 					)
 				}
-
-				utils.AddNodesToCollectionsInRG(ob.meta, rgName, nodes...)
 			}
 		}
 	}
+	if enableRGAutoRecover {
+		utils.RecoverAllCollection(ob.meta)
+	}
```

Comment:
Yes, current implementation we always have to call RecoverAllCollection here to avoid resource group update lost.
After resource group enhancement PR, all node change can be seen at replica_observer.go, those code can be removed.